pkg/controllers/disruption/drift.go

```diff
@@ -36,11 +36,11 @@ import (
 type Drift struct {
 	kubeClient  client.Client
 	cluster     *state.Cluster
-	provisioner *provisioning.Provisioner
+	provisioner *dynamicprovisioning.Provisioner
 	recorder    events.Recorder
 }
 
-func NewDrift(kubeClient client.Client, cluster *state.Cluster, provisioner *provisioning.Provisioner, recorder events.Recorder) *Drift {
+func NewDrift(kubeClient client.Client, cluster *state.Cluster, provisioner *dynamicprovisioning.Provisioner, recorder events.Recorder) *Drift {
```

Comment:
We didn't have any special handling for drift? I think we need to consider being faster for drift for static provisioning -- there are some tenants about drift handling that we can leverage to speed things up, including the fact that if we already are at or below the number of nodes that we intend to have with the static NodePool, we can short-circuit our scheduling logic in favor of just launching a replacement that matches what our static NodePool already looks like